beacon-chain/rpc/prysm/v1alpha1/validator/duties.go

```diff
@@ -123,9 +139,9 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
 			// Next epoch sync committee duty is assigned with next period sync committee only during
 			// sync period epoch boundary (ie. EPOCHS_PER_SYNC_COMMITTEE_PERIOD - 1). Else wise
 			// next epoch sync committee duty is the same as current epoch.
-			nextSlotToEpoch := slots.ToEpoch(s.Slot() + 1)
+			nextEpoch := req.Epoch + 1
```

Comment:
This was wrong, but luckily it was not surfaced because the `old CommitteeAssignments` would mutate the slot. We should use the request epoch instead of the state slot for the query, and only update the next sync committee if the next epoch crosses the boundary as specified below.